nexus/db-lookup/src/lookup.rs

```diff
@@ -257,12 +248,29 @@ impl<'a> LookupPath<'a> {
     }
 
     /// Select a resource of type SiloUser, identified by its id
-    pub fn silo_user_id(self, id: Uuid) -> SiloUser<'a> {
+    pub fn silo_user_id(self, id: SiloUserUuid) -> SiloUser<'a> {
         SiloUser::PrimaryKey(Root { lookup_root: self }, id)
     }
 
+    /// Select a resource of type SiloUser that matches an authenticated Actor
+    pub fn silo_user_actor(
+        self,
+        actor: &'a authn::Actor,
+    ) -> Result<SiloUser<'a>, Error> {
+        match actor.silo_user_id() {
+            Some(silo_user_id) => Ok(SiloUser::PrimaryKey(
+                Root { lookup_root: self },
+                silo_user_id,
+            )),
+
+            None => Err(Error::non_resourcetype_not_found(
+                "could not find silo user",
+            )),
+        }
+    }
+
```

Comment:
I think `match actor` would make it easier to see why this might fail.

Reply:
agreed, done in f926dadc8